### PR TITLE
Add back roll-pitch-yawrate control in GUIDED(NO-GPS) mode

### PR DESCRIPTION
**Pull request by lmark1**
This PR adds back the ability to send body frame roll-pitch and yawrate commands with SET_ATTITUDE_TARGET message bitmask  set to 3 (IGNORE_PITCH_RATE + IGNORE_ROLL_RATE). 

Relevant issue https://github.com/ArduPilot/ardupilot/issues/20444.

### CODE REVIEW

**Review by lthall — COMMENTED** [review]
This does not match the SET_ATTITUDE_TARGET message as the yaw angle as defined in the quaternion is ignored.

**Comment by lmark1** [comment]
I understand that it does not strictly abide by the message definition. 

However, this functionality was very useful for us throughout the years and I would argue that adding it (back) in, along with other cases such as attitude (+feed-forward body rate) and complete body rate control, only makes for a more complete interface with the low-level attitude control.

**Comment by lthall** [comment]
>> I understand that it does not strictly abide by the message definition.

I would say it makes no attempt to abide by the message definition and would impact the very significant number of people that do use this message as it is designed to be used.

>> However, this functionality was very useful for us throughout the years and I would argue that adding it (back) in, along with other cases such as attitude (+feed-forward body rate) and body rate control, only makes for a more complete interface with the low-level attitude control.

As discussed in a few places now, the solution to this problem is to propose a message that can be added to mavlink that we can use properly to provide this functionality.

Some things to consider in the specification of a message would be:
Base it on euler angles and rates where you can only specify one or the other on each axis
Provide climb rate OR thrust command

The other thing I have to highlight when ever I see this. It takes very little effort to integrate the desired yaw rate and calculate the correct target quaternion for the message to implement roll-pitch angle and yaw rate commands using the SET_ATTITUDE_TARGET.

The other interesting point is the usefulness of roll-pitch angle and yaw rate is very limited. The primary use of this is human control.

So to summarise:

- I won't support the corruption of the SET_ATTITUDE_TARGET message,
- I would support the creation of a new message,
- To create a new message there needs to be sufficient community interest to clearly define the applications to be targeted and the range of functionality required for those applications,
- Once the functionality requirements are clear the message specification can be determined.

So far I have not seen this interest. I believe this is because it is easier to just use SET_ATTITUDE_TARGET as it is now as it provides all the functionality required and more.

**Comment by ajshank** [comment]
If I may offer my two cents and address two main points:

> The other interesting point is the usefulness of roll-pitch angle and yaw rate is very limited. The primary use of this is human control.

I think the amount of discussion around this feature suggests otherwise. There is a good amount of literature too that uses R/P angle + Y-rate control, and removing this feature has caught at least some of the users unawares.

>  It takes very little effort to integrate the desired yaw rate and calculate the correct target quaternion for the message [..]

Yes, that is correct, but you've missed the key point (just as before in the issue linked above) -- I could be using an external estimator and its yaw estimate needn't agree with Copter's. Body-frame control/maths will work in spite of this fixed-frame disagreement, and that's what's missing now.


I agree that the current message structure doesn't support this feature, and hence, randomly changing it can be a disaster. My current solution is to add a `GUID_OPTIONS` field [to explicitly enable this feature](https://github.com/ajshank/ardupilot/commit/1c9c2fc89fd4ea319f6d8209dc4f605a51752fd1). If "climb rate" behaviour can be supported this way, I see no reason why yawrate cannot. For the time being, I'll continue to rebase this atop new releases.
